### DISCUSSION TITLE
optimizer

### DISCUSSION THREAD

**Discussion by blockchainPr** [NomicFoundation/hardhat#5408] [Troubleshooting]
Hi . 
i have a smart contract that is of big size so used an optimizer , the contract got compiled but at deployment, i got error, i hope some one can help. i am attaching a screenshot that explains all what i have done.
thanks alot
![Screenshot (1919)](https://github.com/NomicFoundation/hardhat/assets/162696927/01515de1-3aec-492d-93ef-2fbf640393ab)

**Comment by kanej** [comment]
There is likely a syntax error in your `./scripts/deploy.js` file. I would fix up that script file and check it is valid javascript syntax (the error appears to be around line 26).

**Reply by blockchainPr** [reply]
thank you for your answer, however, i looked into my deploy.js file but couldn't identify an error ( kindly see the screenshot)
![Screenshot (1920)](https://github.com/NomicFoundation/hardhat/assets/162696927/b3750767-d56b-4262-a4e6-3a13e3a72083)

**Reply by kanej** [reply]
Have you saved the deploy, contract and config files before running the deploy script?
The cli runs against the files as they are on disk, rather than as they have been edited in vscode.

**Reply by blockchainPr** [reply]
thanks , i re-saved them and it started working but i am getting an error deploying the 3rd smart contract which imports the 1st and the 2nd smart contracts . the error is in the constructor . i need to insert 3 arguments (unit256, address 1st SC, address 2nd SC) . i am attaching a screenshot of the error and the constructor of the 3rd smart contract. kindly help, many thanks
![Screenshot (1921)](https://github.com/NomicFoundation/hardhat/assets/162696927/255d896e-524f-46f6-a77d-bcff7be65ea6)
![Screenshot (1922)](https://github.com/NomicFoundation/hardhat/assets/162696927/2c5b2b1b-c9bb-4724-b04d-1d52828c0f8b)
